sdk/resource/example_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+func ExampleNew() {
+	res, err := resource.New(
+		context.Background(),
+		resource.WithFromEnv(),      // Pull attributes from OTEL_RESOURCE_ATTRIBUTES and OTEL_SERVICE_NAME environment variables.
+		resource.WithTelemetrySDK(), // Provide information about the OpenTelemetry SDK used.
+		resource.WithProcess(),      // Discover and provide process information.
+		resource.WithOS(),           // Discover and provide OS information.
+		resource.WithContainer(),    // Discover and provide container information.
+		resource.WithHost(),         // Discover and provide information.
+		resource.WithAttributes(attribute.String("foo", "bar")), // Add custom resource attributes.
+		// resource.WithDetectors(thirdparty.Detector{}), // Bring your own external Detector implementation.
+	)
+	if errors.Is(err, resource.ErrPartialResource) || errors.Is(err, resource.ErrSchemaURLConflict) {
+		log.Println(err) // Log non-fatal issues.
+	} else if err != nil {
+		log.Fatalln(err) // The error may be fatal.
+	}
+
+	// Now, you can use the resource and e.g. pass it to a tracer or meter provider.
+	fmt.Println(res.SchemaURL())
+
+	// Output: https://opentelemetry.io/schemas/1.24.0
```

Comment:
This is going to add a breaking test for every semconv upgrade. Do we need to output anything here, and if so can it be a stable value?

Reply:
1. I think it is a very good sanity test to make sure that we have not missed to bump the semconv bump in some detector.
2. Adding `// Output:` makes this an [testable example](https://go.dev/blog/examples). Without it the test runner will not execute this code.

Reply:
I would rather not introduce a change detector here.

Reply:
> I would rather not introduce a change detector here.

I do not understand.

The test may find a problem if we do not to use the same `semconv.SchemaURL` in these detectors.

If you prefer I can remove this line.

Reply:
> If you prefer I can remove this line.

:+1: